### ISSUE TITLE
Retarding exhaust VVT which drives DI pump causes issues

### ISSUE THREAD

**Issue by ElDominio** [rusefi/rusefi#5596]
**Describe the bug**
An exhaust VVT that drives to retard, and the base pump angle is 10 degrees, any time you target less than -10 degrees of VVT retard, GDI pump angle breaks.

**Hardware**
4chan
PB Adapter
GDi4

**To Reproduce**
1. set up VVT as normal
2. control exhaust VVT
3. set target angle to under -10 (in this case)
4. watch gdi pump angle go crazy

**Expected behavior**
GDi angle moves consistently

**Screenshots**
No screenshots but log at
https://rusefi.com/online/view.php?log=1297

check GDi pump angle and exhaust VVT angle (GDi pump is driven by exhaust cam)

**Comment by rusefillc** [comment]
problem: negative enginePhase argument would cause AngleBasedEvent to get forever stuck in TriggerSchedule

on the other hand all lobes share the same instance of AngleBasedEvent and we would repurpose that invalid AngleBasedEvent with a more valid one

lastly potentially we have yet another issue - could it be that we keep scheduling different phaseEngle for each lobe while never actually having a chance to fire that event?

**Comment by ElDominio** [comment]
<img width="776" alt="image" src="https://github.com/rusefi/rusefi/assets/5685108/084d746b-55ab-45bb-b029-2f7aa561f3d5">

At 0 degrees of exhaust VVT, next GDi pump angle target looks to be reasonable.

<img width="706" alt="image" src="https://github.com/rusefi/rusefi/assets/5685108/a6f1762d-b592-4103-9574-efe1882df631">

At -20, looks confused, target is mostly flat?

<img width="922" alt="image" src="https://github.com/rusefi/rusefi/assets/5685108/eaba26b1-40aa-4949-ae71-41c9218e9a33">


At -15, angle changes, but it does not follow the recognizable pattern

**Comment by ElDominio** [comment]
https://rusefi.com/online/view.php?log=1305

Still the same issue, no general change

**Comment by rusefillc** [comment]
![image](https://github.com/rusefi/rusefi/assets/48498823/e688165c-be44-4ce4-a990-c157845b0353)

**Comment by rusefillc** [comment]
Depends on #5640

**Comment by rusefillc** [comment]
we believe -50 lobe peak to be more correct? better? red kia value

![image](https://github.com/rusefi/rusefi/assets/48498823/bc4f0410-44c2-497b-b6f4-4cb631fed2d7)

**Comment by rusefillc** [comment]
way to make it bad

![image](https://github.com/rusefi/rusefi/assets/48498823/39797859-5093-476c-b661-a22088d4f1ab)
